src/SuperSelectField.js

```diff
@@ -337,8 +350,18 @@ class SelectField extends Component {
   }
 
   closeMenu = (reason) => {
-    const { onChange, name } = this.props
-    onChange(this.state.selectedItems, name)
+    const { onChange, name, allowSelectAll, selectAllItem } = this.props
+    let { selectedItems } = this.state
+    if (allowSelectAll) {
+      const selectAllItemIndex = selectedItems.findIndex(item => item.value === selectAllItem.value);
+      if (selectAllItemIndex !== -1) {
+        selectedItems = [
```

Comment:
You are mutating the state here, best practice is to first make a copy then play with it.

Also i don't understand what are you trying to achieve with these consecutive slices ?
Exclude the `selectAllItem` ? if yes, just use `.filter()`